weaver/processes/wps_package.py

```diff
@@ -2228,29 +2228,40 @@ def _update_status_dispatch(_provider, _message, _progress, _status):
         app_hints = list(filter(lambda h: any(h["class"].endswith(t) for t in CWL_REQUIREMENT_APP_TYPES), all_hints))
         if len(app_hints) > 1:
             raise ValueError("Package 'requirements' and/or 'hints' define too many conflicting values: {}, "
-                             "only one permitted amongst {}.".format(app_hints, list(CWL_REQUIREMENT_APP_TYPES)))
+                             "only one permitted amongst {}.".format(list(app_hints), CWL_REQUIREMENT_APP_TYPES))
         requirement = app_hints[0] if app_hints else {"class": ""}
-        req_class = requirement["class"]
-        if req_class.endswith(CWL_REQUIREMENT_APP_WPS1):
-            LOGGER.info("WPS-1 Package resolved from requirement/hint: %s", req_class)
-            req_params = ["provider", "process"]
-            if not all(req in requirement for req in ["provider", "process"]):
-                raise ValueError("Missing requirement [{}] details amongst {}".format(req_class, req_params))
-            provider = requirement["provider"]
-            # The process id of the provider isn't required to be the same as the one use in the EMS
-            process = requirement["process"]
+
+        def _get_wps1_params(_requirement):
+            params = {}
+
+            required_params = ["provider", "process"]
+            for param in required_params:
+                if param not in _requirement:
+                    raise ValueError("Missing requirement detail [{}]: {}".format(_requirement["class"], param))
+                params[param] = _requirement[param]
+            return params
```

Comment:
move to separate function (not within) 
also rename more specifically since it is not limited to WPS-1

Reply:
Also update docs for WPS1/ESGF_CWT type of process.   (docs/processes.rst)
This kind of details are hard to find without any reference/sample.

Reply:
would it need to be used outside of `get_job_process_definition` ?
and for the doc, sample in/out of this utility ?

Reply:
To allow a process definition to be parsed correctly, there is a strong requirement of CWL definition to have something like so: 
```
hints: 
  WPS1Requiement | ESGF-CWTRequiement: 
    provider: <url>
    process: <id>
```
In the case of WPS-1, is was generally auto-filled by weaver here: 
https://github.com/crim-ca/weaver/blob/ed55a73cda8b85574242fac400c525a9fcf73233/weaver/processes/wps_package.py#L1575-L1580

But I would like to have this documented more explicitly since this is not done for ESGF use-case and because it is something very Weaver-specific that cannot be guessed easily by a user. 

The function can be left there for now. 
If needed, we will move it out.

Reply:
Yes, good idea to document this particularity
I have added some stuff in the docs, to explain that for ESGF-CWT it is handled differently

Reply:
If you find other important notes or details that would help any future user to work with WPS-1 or ESGF processes or workflows, feel free to suggests more additions to documentation. 
I'm particularly interested in things that are not obvious like the above, or other frustrations that you encountered that could help guide new users with such deployment/execute definitions.